var/spack/repos/builtin/packages/py-jedi/package.py

```diff
@@ -12,9 +12,17 @@ class PyJedi(PythonPackage):
     homepage = "https://github.com/davidhalter/jedi"
     url      = "https://pypi.io/packages/source/j/jedi/jedi-0.9.0.tar.gz"
 
-    # unfortunately pypi.io only offers a .whl
+    version('0.13.3', sha256='2bb0603e3506f708e792c7f4ad8fc2a7a9d9c2d292a358fbbd58da531695595b')
+    # unfortunately pypi.io only offers a .whl for 0.10.0
     version('0.10.0', sha256='d6a7344df9c80562c3f62199278004ccc7c5889be9f1a6aa5abde117ec085123',
                 url='https://github.com/davidhalter/jedi/archive/v0.10.0.tar.gz')
     version('0.9.0', sha256='3b4c19fba31bdead9ab7350fb9fa7c914c59b0a807dcdd5c00a05feb85491d31')
 
-    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('python@2.6:2.7,3.2:3.4', type=('build', 'run'), when='@0.9.0')
+    depends_on('python@2.6:2.7,3.3:3.5', type=('build', 'run'), when='@0.10.0')
+    depends_on('python@2.7,3.4:3.7', type=('build', 'run'), when='@0.13.3')
```

Comment:
I never add upper version ranges for Python 3 unless the build script explicitly crashes. Most Python 3.4-compatible packages are also compatible with Python 3.8, it's just that it wasn't out at the time to test it.
```suggestion
    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'), when='@0.13.3')
```